src/merkle/path.rs

```diff
@@ -189,6 +191,55 @@ pub struct RootPath {
     pub path: MerklePath,
 }
 
+// SERILIZATION
+// ================================================================================================
+impl Serializable for MerklePath {
+    fn write_into<W: winter_utils::ByteWriter>(&self, target: &mut W) {
+        target.write_u64(self.nodes.len() as u64);
```

Comment:
We probably don't need to support Merkle paths longer than 256 - so, we could store the length in `u8` (but we'll probably need to make this assumption explicit in the constructors - maybe via `assert!`?)

Reply:
I'm going to add the assert as requested. However, I think this is bad API, and we should return a `Result` instead. Not doing this since that is a breaking change and will require changing at least 4 repos (with this included).

Reply:
As a side note, I tend to favor security/safety over performance, this is a good example of that, IMO we should avoid pushing code that can panic downstream, since that makes evaluating the correctness of the node much harder. Here however, performance is being prioritize. It would be good to define what are the priorities so that for future PRs I know which approach to take.

Reply:
> Here however, performance is being prioritize.

AFAIK asserting that the same performance characteristics as using `?`, unless I'm missing something? They both run the "error check", and either panic/unwind, or initiate a "function return". Wrapping in a `Result` is also trivial and doesn't affect performance.

So I agree with you, we should *always* be explicit about how a function can fail and return a `Result` instead of using `assert`s.

Reply:
Generally, I agree and here it was done more for ergonomic reasons and to avoid breaking changes rather than performance. I also think that panics in the code which is not expected to be called publicly are OK (as long as they are well documented).

So, I think there are two options to deal with this:
- Make `MerklePath` constructor return `Result`.
- Make `MerklePath` constructor private to this crate.

Should we create an issue to address this in the future?

Reply:
https://github.com/0xPolygonMiden/crypto/issues/233